tasks/kernel_matrix_testing/gdb.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from invoke.context import Context
+
+from tasks.kernel_matrix_testing import stacks
+from tasks.kernel_matrix_testing.infra import LibvirtDomain, build_infrastructure
+from tasks.kernel_matrix_testing.platforms import get_platforms
+from tasks.kernel_matrix_testing.tool import Exit, info
+from tasks.kernel_matrix_testing.vars import KMTPaths
+from tasks.libs.types.arch import Arch
+
+if TYPE_CHECKING:
+    from tasks.kernel_matrix_testing.types import (
+        Component,  # noqa: F401
+        KMTArchNameOrLocal,
+    )
+
+
+class KernelVersion:
+    def __init__(self, kv: str):
+        self.major = kv.split(".")[0]
+        self.minor = kv.split(".")[1]
+        self.patch = kv.split(".")[2]
+
+    def __str__(self):
+        return f"{self.major}.{self.minor}.{self.patch}"
+
+    def __eq__(self, other: object):
+        if isinstance(other, KernelVersion):
+            return self.major == other.major and self.minor == other.minor and self.patch == other.patch
+
+        raise NotImplementedError
+
+    def __lt__(self, other: KernelVersion) -> bool:
+        if not isinstance(other, KernelVersion):
+            raise NotImplementedError
+
+        if self.major < other.major:
+            return True
+        if self.major == other.major and self.minor < other.minor:
+            return True
+
+        return self.minor == other.minor and self.patch < other.patch
+
+    def __lte__(self, other: KernelVersion) -> bool:
+        return self.__lt__(other) or self.__eq__(other)
+
+    def __gt__(self, other: KernelVersion) -> bool:
+        return not self.__lte__(other)
+
+
+class GDBPaths:
+    def __init__(self, vm_tag: str, image_version: str, stack: str, arch: KMTArchNameOrLocal):
+        self.tag = vm_tag
+        self.image_version = image_version
+        self.kmt_paths = KMTPaths(stack, Arch.from_str(arch))
+
+    @property
+    def vmlinux(self):
+        return self.kmt_paths.gdb / self.tag / self.image_version / "vmlinux.dbg"
+
+    @property
+    def kernel_source(self):
+        return self.kmt_paths.gdb / self.tag / self.image_version / "kernel-source"
+
+
+class UbuntuGDBProvision:
+    def __init__(self, vm: LibvirtDomain, image_version: str, kernel: str):
+        self.target = vm
+        self.image_version = image_version
+        self.kernel = KernelVersion(kernel.split('-')[0])
+
+    def run(self, ctx: Context, stack: str):
+        self.target.copy(ctx, "tasks/kernel_matrix_testing/provision/ubuntu-dbg.sh", "/tmp/provision.sh")
```

Comment:
```suggestion
        self.target.copy(ctx, get_repo_root() / "tasks/kernel_matrix_testing/provision/ubuntu-dbg.sh", "/tmp/provision.sh")
```

Just in case some runs from another path.